site/content/docs/0.2/extend/icons.md

```diff
@@ -1,11 +1,736 @@
+<svg width="3rem" height="3rem" aria-hidden="true">
+<use xlink:href="/docs/{{< param docs_version >}}/assets/img/ouds-web-sprite.svg#settings"/>
+</svg>
+<svg width="3rem" height="3rem" aria-hidden="true">
+<use xlink:href="/docs/{{< param docs_version >}}/assets/img/ouds-web-sprite.svg#trash"/>
+</svg>
+{{< /example >}}
+
+To speed up loading, the SVG sprite file can be preloaded in the `<head>` of the page:
+
+{{< example lang="html" show_preview="false" >}}
+<link rel="preload" href="/path/to/your/sprite.svg" as="image" type="image/svg+xml">
+{{< /example >}}
+
+### Web font
+
+Web font is not the preferred solution. However, for websites needing more than 20 icons, you may consider using a Web font instead of an SVG sprite.
+
+Using the [Solaris icons finder]({{< param icons >}}), you can generate a Web font containing all your icons and use it like a classic font. Web font icons can be styled through CSS properties like `font-size` and `color`. please keep in mind that adding or removing icons from the font afterward can be complicated.
+
+{{< example >}}
+<span class="si si-settings" aria-hidden="true" style="font-size: 3rem"></span>
+{{< /example >}}
+
+
+<!--{{< callout warning >}}
+Note that with this technique, you cannot have multicolor icons, so you cannot render the [warning icon]({{< docsref "extend/icons#warning-icon" >}}).
+{{< /callout >}}-->
+
+### Inline SVG
+
+This technique should only be used if you have few icons to render, and if they are used only once in your website.
+
+You can embed your icons directly within the HTML of your page (as opposed to an external image file). This way to use SVGs can benefit of the power of `currentColor` for easy coloring and theming.
+
+{{< example class="mt-0" >}}
+<svg fill="currentColor" class="decorative-sm-icon" viewBox="0 0 1000 1000" aria-hidden="true">
+  <path d="M656.7 422.409a229.96 229.96 0 0 1-315.39.008A224.95 224.95 0 0 0 224.064 615H224v210a100 100 0 0 0 100 100h450V620a224.94 224.94 0 0 0-117.3-197.591M679 255A180 180 0 1 1 499 75a180 180 0 0 1 180 180" style="fill-rule:evenodd"/>
+</svg>
+{{< /example >}}
+
+### SVG external image
+
+You can use it when:
+- you only have few icons to render
+- you don't need to change the icons colors. For that reason, **we do not recommend to use that technique**, so it won't be documented here.
+
+### CSS background SVG
+
+CSS background icons should be used when you can't or don't want to refer to an external image, or if you want to include the icon in a CSS `::before` or `::after` pseudo-element.
+
+You can use the SVG code within your CSS (be sure to escape any characters with [our internal `escape-svg()` function]({{< docsref "/customize/sass" >}}#escape-svg)). When no dimensions are specified via `width` and `height` on the `<svg>`, the icon will fill the available space. CSS background SVGs can be styled through CSS properties like `width`, `height` or `background-color`. Note that the `xmlns` attribute is required.
```

Comment:
This doesn't look that true for me, maybe we should introduce a part about the not sustainaibility when the theme context can be changed ?

Reply:
which part, can you be more specific ?